velodrome/transform/plugins/bundle.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+)
+
+// ByDuration sorts a slice of time.Duration
+type ByDuration []time.Duration
+
+func (b ByDuration) Len() int           { return len(b) }
+func (b ByDuration) Less(i, j int) bool { return b[i] < b[j] }
+func (b ByDuration) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+
+// BundledStates saves the state of multiple issues/pull-requests
```

Comment:
Based on the comment I'm a little confused about why this is necessary vs just keeping a `map[string]State` wherever you need it. It looks like there's much more to this though, like `Total` and `Percentile` stuff.